src/ompl/geometric/planners/bitstar/BITstar.h

```diff
@@ -0,0 +1,614 @@
+            void pruneSamples();
+
+            /** \brief Checks an edge for collision. A wrapper to SpaceInformation->checkMotion that tracks number of collision checks. */
+            bool checkEdge(const vertex_pair_t& edge);
+
+            /** \brief Actually remove a sample from its NN struct: */
+            void dropSample(VertexPtr oldSample);
+
+            /** \brief Add an edge from the edge queue to the tree. Will add the state to the vertex queue if it's new to the tree or otherwise replace the parent. Updates solution information if the solution improves. */
+            void addEdge(const vertex_pair_t& newEdge, const ompl::base::Cost& edgeCost, const bool& removeFromFree, const bool& updateDescendants);
+
+            /** \brief Replace the parent edge with the given new edge and cost */
+            void replaceParent(const vertex_pair_t& newEdge, const ompl::base::Cost& edgeCost, const bool& updateDescendants);
+
+            /** \brief The special work that needs to be done to update the goal vertex is the solution has changed. */
+            void updateGoalVertex();
+
+            /** \brief Add a sample */
+            void addSample(const VertexPtr& newSample);
+
+            /** \brief Add a vertex to the graph */
+            void addVertex(const VertexPtr& newVertex, const bool& removeFromFree);
+
+            /** \brief Get the nearest samples from the freeStateNN_ using the appropriate "near" definition (i.e., k or r). */
+            void nearestSamples(const VertexPtr& vertex, std::vector<VertexPtr>* neighbourSamples);
+
+            /** \brief Get the nearest samples from the vertexNN_ using the appropriate "near" definition (i.e., k or r). */
+            void nearestVertices(const VertexPtr& vertex, std::vector<VertexPtr>* neighbourVertices);
+            ///////////////////////////////////////////////////////////////////
+
+            ///////////////////////////////////////////////////////////////////
+            //Helper functions for sorting queues/nearest-neighbour structures and the related calculations.
+            /** \brief The distance function used for nearest neighbours. Calculates the distance directionally from the given state to all the other states (can be used on states either in our out of the graph).*/
+            double nnDistance(const VertexPtr& a, const VertexPtr& b) const;
+            ///////////////////////////////////////////////////////////////////
+
+            ///////////////////////////////////////////////////////////////////
+            //Helper functions for various heuristics.
+            /** \brief Calculates a heuristic estimate of the cost of a solution constrained to pass through a vertex, independent of the current cost-to-come. I.e., combines the heuristic estimates of the cost-to-come and cost-to-go. */
+            ompl::base::Cost lowerBoundHeuristicVertex(const VertexPtr& edgePair) const;
```

Comment:
I personally find that "absolute" namespaces make code from large projects easier to understand (especially at a glance)... and I don't mind doing the extra typing. :smile: